### PR TITLE
[WIP] Banshee

### PR DESCRIPTION
**Pull request by zohl**
This set of patches is an attempt to bring banshee, a musical player, into the nix world.

I managed to compile and run it, so the application window appeared. But many features like playing music and syncing devices didn't work. This was a few weeks ago, now under the current version of nixos-unstable the program fails to run at all. It is the point where I'm out of ideas how to make it work.

According to mono debug information (`MONO_LOG_LEVEL=debug`) there are bunch of missing libraries. Yet at the same time in a debian I get almost the same output. Seems like this messages aren't relevant to the problem. The reason why it fails to create window is also not clear for me. Moreover I do not understand well enough how does mono perform searching and linking the required libraries.
So I'm seeking for some advice on this problem.

PS. I'm not a mono/dotnet expert (and have no experience in this sphere), so many things I did to make this work are probably dirty hacks and can be improved. Any ideas and comments are welcomed.

### CODE REVIEW

File: pkgs/development/compilers/boo/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, pkgconfig, dbus, mono, makeWrapper, nant 
+, shared_mime_info, gtksourceview, gtk
+, targetVersion ? "4.5" }:
+
+let
+  release = "alpha";
+  version = "2013-10-21";
+in stdenv.mkDerivation {
```
Comment (joachifm) [inline pkgs/development/compilers/boo/default.nix]: Usually, we use `rec` and define `version` within the attrset passed to `mkDerivation`

File: pkgs/development/libraries/dbus-sharp-glib/dbus-sharp-glib-1.0.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  buildInputs = [ pkgconfig autoconf automake mono dbus-sharp-1_0 ];
+ 
+  dontStrip = true;
+
+  patchPhase = ''
+    ./autogen.sh
+  '';
+
+  meta = with stdenv.lib; {
+    description = "D-Bus for .NET: GLib integration module.";
```
Comment (joachifm) [inline pkgs/development/libraries/dbus-sharp-glib/dbus-sharp-glib-1.0.nix]: Please remove the trailing dot in the description

**Comment by jagajaga** [comment]
LGTM

Waiting for the result! Awesome PR.

**Comment by joachifm** [comment]
@jagajaga My understanding is that this doesn't actually work yet

**Comment by jagajaga** [comment]
@joachifm that's why I wrote "waiting for the result".

**Comment by joachifm** [comment]
Oh right, sorry

**Comment by zohl** [comment]
Thank you for the feedback, for the comments and for the patches -- now banshee is able to run and even play music!
There are still some improvements to be done, so this PR should not be merged yet. I'll update it eventually.

File: pkgs/development/libraries/gio-sharp/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  buildInputs = [ mono gtk-sharp ];
+
+  dontStrip = true;
+
+  prePatch = ''
+    ./autogen-2.22.sh
+  '';
+
+  meta = with stdenv.lib; {
+    description = "gio-sharp binds the GIO API, which isn't available in Gtk#";
```
Comment (jagajaga) [inline pkgs/development/libraries/gio-sharp/default.nix]: Please omit pkg name in the description.

File: pkgs/development/libraries/gdata-sharp/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    sed -i -e 's#^\(DEFINES=.*\)\(.\)#\1 /r:third_party/Newtonsoft.Json.dll\2#' Makefile
+    #             carriage return ^
+  '';
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  meta = with stdenv.lib; {
+    homepage = https://code.google.com/archive/p/google-gdata/;
+
+    description = "The Google Data APIs (GData)";
```
Comment (jagajaga) [inline pkgs/development/libraries/gdata-sharp/default.nix]: Please remove `(GData)`.

File: pkgs/development/libraries/mono-addins/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  nativeBuildInputs = [ pkgconfig autoreconfHook ];
+  buildInputs = [ mono gtk-sharp-2_0 ];
+
+  dontStrip = true;
+
+  meta = with stdenv.lib; {
+    homepage = http://www.mono-project.com/archived/monoaddins/;
+
```
Comment (jagajaga) [inline pkgs/development/libraries/mono-addins/default.nix]: Remove extra lines please.

**Comment by zohl** [comment]
@jagajaga, thanks for the review, I've fixed it according to your comments.

**Comment by jagajaga** [comment]
Can I merge this?

**Comment by zohl** [comment]
There are still features that do not work. I would like to fix some of them (especially syncing with ipods) before the patch will be merged. If I do not succeeded with it to the end of the week, feel free to merge -- at least basic features works here.

**Comment by jagajaga** [comment]
I will merge (I want to try this player) and wait for more patches in the future :)

**Comment by zohl** [comment]
Alright, thank you!

**Pull request merged**